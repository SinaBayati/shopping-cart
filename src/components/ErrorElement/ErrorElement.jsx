import { Link } from "react-router-dom";
import styled from "styled-components";

const Wrapper = styled.div`
  background-color: black;
  height: 100vh;
  padding: 1rem;
  color: white;
  font-family: "Gill Sans", "Gill Sans MT", Calibri, "Trebuchet MS", sans-serif;
  font-size: 2rem;
`;

const StyledLink = styled(Link)`
  text-decoration: none;
  color: white;
  display: block;
  font-size: 1.2rem;
  padding: 2rem 0;

  &:visited,
  &:active {
    color: white;
  }
`;

function ErrorElement() {
  return (
    <Wrapper>
      An error has occurred
      <StyledLink to="/">go back to homepage</StyledLink>
    </Wrapper>
  );
}

export default ErrorElement;
