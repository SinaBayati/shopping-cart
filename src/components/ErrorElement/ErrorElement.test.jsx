import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import ErrorElement from "./ErrorElement";

describe("ErrorElement", () => {
  it("renders ErrorElement correctly", () => {
    const { container } = render(<ErrorElement />, { wrapper: BrowserRouter });

    expect(container).toMatchSnapshot();
  });
});
