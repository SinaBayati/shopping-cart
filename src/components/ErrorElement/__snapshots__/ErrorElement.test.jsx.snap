// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorElement > renders ErrorElement correctly 1`] = `
<div>
  <div
    class="sc-Qotzb iIRrMf"
  >
    An error has occurred
    <a
      class="sc-fYsHOw jlChIF"
      data-discover="true"
      href="/"
    >
      go back to homepage
    </a>
  </div>
</div>
`;
